pkg/networkdevice/profile/profiledefinition/yaml_utils.go

```diff
@@ -0,0 +1,47 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package profiledefinition
+
+// StringArray is list of string with a yaml un-marshaller that support both array and string.
+// See test file for example usage.
+// Credit: https://github.com/go-yaml/yaml/issues/100#issuecomment-324964723
+type StringArray []string
```

Comment:
Out of curiosity, why do we allow this? Is this just for backwards compatibility with older config files, where some fields might have been singular instead of arrays?

Reply:
Yes, indeed, it's used for `sysobjectid`. historically `sysobjectid` only accepts a string, but now, it can be also a list.

https://github.com/DataDog/datadog-agent/blob/1df99627c3a1ae605e81ee63a932b60eb9dd49bb/pkg/networkdevice/profile/profiledefinition/profile_definition.go#L18